webapp/packages/core-blocks/src/FormControls/InputAutocompletionMenu.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observer } from 'mobx-react-lite';
+import { useEffect, useRef } from 'react';
+
+import BaseDropdownStyles from '../FormControls/BaseDropdown.module.css';
+import { IconOrImage } from '../IconOrImage.js';
+import { Menu } from '../Menu/Menu.js';
+import { MenuItem } from '../Menu/MenuItem.js';
+import type { IMenuState } from '../Menu/MenuStateContext.js';
+import { s } from '../s.js';
+import { Text } from '../Text.js';
+import { useS } from '../useS.js';
+import style from './InputAutocompletionMenu.module.css';
+import { type InputAutocompleteProposal, type InputAutocompleteStrategy, useInputAutocomplete } from './useInputAutocomplete.js';
+
+interface AutocompletionProps {
+  sourceHints: InputAutocompleteProposal[];
+  inputRef: React.RefObject<HTMLInputElement | HTMLTextAreaElement>;
+  matchStrategy?: InputAutocompleteStrategy;
+  className?: string;
+  onSelect?: (proposal: InputAutocompleteProposal) => void;
+}
+
+export const InputAutocompletionMenu = observer(function InputAutocompletionMenu({
+  sourceHints,
+  className,
+  matchStrategy,
+  inputRef,
+  onSelect,
+}: AutocompletionProps) {
+  const styles = useS(style, BaseDropdownStyles);
+  const menuRef = useRef<IMenuState>();
+  const autocompleteState = useInputAutocomplete(inputRef, {
+    sourceHints,
+    matchStrategy,
+  });
```

Comment:
please move this hook outside from the component, so you can use it and use input ref in the same component